pjmedia/src/pjmedia/vid_conf.c

```diff
@@ -93,6 +93,7 @@ typedef struct vconf_port
     unsigned             idx;           /**< Port index.                    */
     pj_str_t             name;          /**< Port name.                     */
     pjmedia_port        *port;          /**< Video port.                    */
+    pj_bool_t            is_new;        /**< Port newly added?              */
```

Comment:
I'm still not quite clear with the "new port" check. Since it also applies to video, is it considered a bug fix for the current code?

Reply:
Yes, it is kind of bug fix (for some reason, for video it causes less/no harm).

Currently adding a port to conf needs to be done synchronously to avoid premature destroy (if async, the conf may end-up adding/accessing a destroyed port). However, we cannot increment/change `conf->port_cnt` there as it is accessed by conf `get_frame()` without mutex protection, e.g: https://github.com/pjsip/pjproject/blob/ed32a91a4a27fe4ebf3e53478f40a2a18396544d/pjmedia/src/pjmedia/conference.c#L2387
So, the "new port" check is introduced to be able to increment the `conf->port_cnt` safely.